src/metric/elementwise_metric.cu

```diff
@@ -45,7 +45,7 @@ namespace {
 template <typename Fn>
 PackedReduceResult Reduce(Context const* ctx, MetaInfo const& info, Fn&& loss) {
   PackedReduceResult result;
-  auto labels = info.labels.View(ctx->Device());
+  auto labels = info.labels.View(ctx->Device().IsSycl() ? DeviceOrd::CPU() : ctx->Device());
```

Comment:
I assume you want to use `IsCPU` instead of `IsSycl`?

Reply:
No. This function doesn't have sycl-specific implementation yet. So for the proper execution on `cpu `the data should be transferred back to host.
I plan to add sycl-implementation for it in the future. Didn't do it by single PR just to avoid unreadable PR size.